### PR TITLE
Stop relying on name as core.data_source_id

### PR DESCRIPTION
**Pull request by spolu**
## Description

- `core` generates a unique ID for `data_source_id` instead of receiving it from `front`
- `front` stop sending name (store the received Id in dustAPIDataSourceId)
- improve handling of core retrieved data_source_id in document tracker

Migration to move all core data_source_id to generated IDs will be tackled in a follow-up PR. Everything works properly for now since core data_source_id is stored in dustAPIDataSourceId on front side.

Tests:
- [x] create a data source with core deployed but not front
- [x] create a folder
- [x] create a connection
- [x] check upsertion works
- [x] check retrieval works on both new and old data sources

r? @Fraggle cc @flvndvd 

Follow-up work: migration to move all core.data_source_id to a generated Id and update dustAPIDataSourceId

## Risk

Low once properly tested

## Deploy Plan

- deploy `core` (first)
- deploy `front`
- run migration

### CODE REVIEW

File: core/src/data_sources/data_source.rs [hunk 1]
```diff
@@ -941,7 +941,6 @@ impl DataSource {
                 payload.insert("timestamp", document.timestamp as i64);
                 payload.insert("chunk_offset", c.offset as i64);
                 payload.insert("chunk_hash", c.hash.clone());
-                payload.insert("data_source_id", self.data_source_id.clone());
```
Comment (flvndvd) [inline core/src/data_sources/data_source.rs:944]: Why don't we want to keep the `data_source_id` in Qdrant?

File: front/lib/resources/storage/models/data_source.ts [hunk 2]
```diff
@@ -95,6 +95,7 @@ DataSourceModel.init(
       { fields: ["workspaceId", "name"], unique: true },
       { fields: ["workspaceId", "connectorProvider"] },
       { fields: ["workspaceId", "vaultId"] },
+      { fields: ["dustAPIProjectId"] },
```
Comment (flvndvd) [inline front/lib/resources/storage/models/data_source.ts:98]: Where is the migration?

**Review by flvndvd — COMMENTED** [review]
LGTM, left two comments.

File: front/lib/resources/storage/models/data_source.ts (hunk 2 above)
Comment (spolu) [reply front/lib/resources/storage/models/data_source.ts:98]: pushed :+1:

File: core/src/data_sources/data_source.rs (hunk 1 above)
Comment (spolu) [reply core/src/data_sources/data_source.rs:944]: First it serves no purpose. Second if we wanted to keep it we would have to migrate it which is very challenging

Comment (flvndvd) [reply core/src/data_sources/data_source.rs:944]: For sure we will need it in the future to apply proper data segregation!

Comment (spolu) [reply core/src/data_sources/data_source.rs:944]: This is the purpose `data_source_internal_id` as per the slack thread.

File: front/migrations/db/migration_92.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+-- Migration created on Sep 26, 2024
+CREATE INDEX "data_sources_dust_api_project_id" ON "data_sources" ("dustAPIProjectId");
```
Comment (Fraggle) [inline front/migrations/db/migration_92.sql:2]: nit: CONCURRENTLY ? (even if locally it doesn't really matter)

**Review by Fraggle — COMMENTED** [review]
Sorry a bit late but LGTM.

**Pull request merged**